src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt

```diff
@@ -291,18 +302,30 @@ open class RsDefaultInsertHandler : InsertHandler<LookupElement> {
             is RsStructItem -> appendSemicolon(context, curUseItem)
 
             is RsFunction -> {
-                if (curUseItem != null) {
-                    appendSemicolon(context, curUseItem)
-                } else {
-                    val isMethodCall = context.getElementOfType<RsMethodOrField>() != null
-                    if (!context.alreadyHasCallParens && !context.isInMetaItem) {
-                        document.insertString(context.selectionEndOffset, "()")
-                        context.doNotAddOpenParenCompletionChar()
+                when {
+                    curUseItem != null -> {
+                        appendSemicolon(context, curUseItem)
                     }
-                    val caretShift = if (element.valueParameters.isEmpty() && (isMethodCall || !element.hasSelfParameters)) 2 else 1
-                    EditorModificationUtil.moveCaretRelatively(context.editor, caretShift)
-                    if (element.valueParameters.isNotEmpty()) {
-                        AutoPopupController.getInstance(element.project)?.autoPopupParameterInfo(context.editor, element)
+                    element.isProcMacroDef -> {
+                        if (element.isBangProcMacroDef && !context.nextCharIs('!')) {
+                            document.insertString(context.selectionEndOffset, "!()")
+                            EditorModificationUtil.moveCaretRelatively(context.editor, 2)
+                        }
+                        if (element.isAttributeProcMacroDef && context.nextCharIs(']')) {
+                            EditorModificationUtil.moveCaretRelatively(context.editor, 1)
+                        }
```

Comment:
I don't think we should do it since a user may want to insert some argument then, e.g. `#[foo(bar)]`